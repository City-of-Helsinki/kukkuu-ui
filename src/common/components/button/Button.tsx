import * as React from 'react';
import { Button as HdsButton, ButtonProps as HDSButtonProps } from 'hds-react';

type ButtonProps = Omit<HDSButtonProps, 'variant'> & {
  variant?: HDSButtonProps['variant'] | 'disabled';
};

const kukkuuSecondaryButtonStyles = {
  '--background-color': 'var(--color-white)',
  '--background-color-hover': 'var(--color-white)',
  '--background-color-focus': 'var(--color-white)',
  '--background-color-hover-focus': 'var(--color-white)',
};
const kukkuuSupplementaryButtonStyles = {
  '--background-color': 'transparent',
  '--background-color-hover': 'transparent',
  '--background-color-focus': 'transparent',
  '--background-color-hover-focus': 'transparent',
  '--border-color': 'none',
};
const kukkuuDisabledButtonStyles = {
  '--background-color': 'var(--color-black-20)',
  '--background-color-hover': 'var(--color-black-30)',
  '--background-color-focus': 'var(--color-black-40)',
  '--background-color-hover-focus': 'var(--color-black-40)',
  '--border-color': 'var(--color-black-20)',
  '--border-color-hover': 'var(--color-black-30)',
  '--border-color-hover-focus': 'var(--color-black-40)',
  '--border-color-focus': 'var(--color-black-40)',
};

const kukkuuButtonStyles = {
  '--color': 'var(--color-black)',
  '--color-hover': 'var(--color-black)',
  '--color-focus': 'var(--color-black-90)',
  '--color-hover-focus': 'var(--color-black-90)',
  '--background-color': 'var(--color-summer)',
  '--background-color-hover': 'var(--color-summer-dark)',
  '--background-color-focus': 'var(--color-summer)',
  '--background-color-hover-focus': 'var(--color-summer-dark)',
  '--background-selected': 'var(--color-summer)',
  '--border-color': 'var(--color-summer)',
  '--border-color-hover': 'var(--color-summer-dark)',
  '--border-color-focus': 'var(--color-summer-dark)',
  '--border-color-hover-focus': 'var(--color-summer-dark)',
  '--border-color-selected': 'var(--color-summer)',
  '--border-color-selected-hover': 'var(--color-summer-dark)',
  '--border-color-selected-focus': 'var(--color-summer)',
  '--border-color-selected-hover-focus': 'var(--color-summer)',
} as React.CSSProperties;

const Button = ({
  style = kukkuuButtonStyles,
  className,
  variant,
  children,
  ...rest
}: ButtonProps) => {
  switch (variant) {
    case 'secondary':
      style = {
        ...kukkuuButtonStyles,
        ...kukkuuSecondaryButtonStyles,
      };
      break;
    case 'supplementary':
      style = {
        ...kukkuuButtonStyles,
        ...kukkuuSupplementaryButtonStyles,
      };
      break;
    case 'disabled':
      // Can be used to set the button to appear disabled without
      // actually disabling it.
      style = {
        ...kukkuuButtonStyles,
        ...kukkuuDisabledButtonStyles,
      };
      break;
    default:
  }

  return (
    <HdsButton
      className={className}
      style={style}
      children={children}
      {...rest}
    ></HdsButton>
  );
};

export default Button;
