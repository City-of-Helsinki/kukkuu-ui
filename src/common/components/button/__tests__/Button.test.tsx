import { shallow } from 'enzyme';

import Button from '../Button';

it('renders snapshot correctly', () => {
  const button = shallow(<Button>foo</Button>);
  expect(button.html()).toMatchSnapshot();
});
