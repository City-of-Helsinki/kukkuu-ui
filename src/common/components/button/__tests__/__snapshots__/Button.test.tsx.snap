// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders snapshot correctly 1`] = `
<div>
  <button
    class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO"
    style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: var(--color-summer); --background-color-hover: var(--color-summer-dark); --background-color-focus: var(--color-summer); --background-color-hover-focus: var(--color-summer-dark); --background-selected: var(--color-summer); --border-color: var(--color-summer); --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
    type="button"
  >
    <span
      class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
    >
      foo
    </span>
  </button>
</div>
`;
